{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "91bee0e4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "            FastQC - A high throughput sequence QC analysis tool\n",
      "\n",
      "SYNOPSIS\n",
      "\n",
      "\tfastqc seqfile1 seqfile2 .. seqfileN\n",
      "\n",
      "    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] \n",
      "           [-c contaminant file] seqfile1 .. seqfileN\n",
      "\n",
      "DESCRIPTION\n",
      "\n",
      "    FastQC reads a set of sequence files and produces from each one a quality\n",
      "    control report consisting of a number of different modules, each one of \n",
      "    which will help to identify a different potential type of problem in your\n",
      "    data.\n",
      "    \n",
      "    If no files to process are specified on the command line then the program\n",
      "    will start as an interactive graphical application.  If files are provided\n",
      "    on the command line then the program will run with no user interaction\n",
      "    required.  In this mode it is suitable for inclusion into a standardised\n",
      "    analysis pipeline.\n",
      "    \n",
      "    The options for the program as as follows:\n",
      "    \n",
      "    -h --help       Print this help file and exit\n",
      "    \n",
      "    -v --version    Print the version of the program and exit\n",
      "    \n",
      "    -o --outdir     Create all output files in the specified output directory.\n",
      "                    Please note that this directory must exist as the program\n",
      "                    will not create it.  If this option is not set then the \n",
      "                    output file for each sequence file is created in the same\n",
      "                    directory as the sequence file which was processed.\n",
      "                    \n",
      "    --casava        Files come from raw casava output. Files in the same sample\n",
      "                    group (differing only by the group number) will be analysed\n",
      "                    as a set rather than individually. Sequences with the filter\n",
      "                    flag set in the header will be excluded from the analysis.\n",
      "                    Files must have the same names given to them by casava\n",
      "                    (including being gzipped and ending with .gz) otherwise they\n",
      "                    won't be grouped together correctly.\n",
      "                    \n",
      "    --nano          Files come from nanopore sequences and are in fast5 format. In\n",
      "                    this mode you can pass in directories to process and the program\n",
      "                    will take in all fast5 files within those directories and produce\n",
      "                    a single output file from the sequences found in all files.                    \n",
      "                    \n",
      "    --nofilter      If running with --casava then don't remove read flagged by\n",
      "                    casava as poor quality when performing the QC analysis.\n",
      "                   \n",
      "    --extract       If set then the zipped output file will be uncompressed in\n",
      "                    the same directory after it has been created.  By default\n",
      "                    this option will be set if fastqc is run in non-interactive\n",
      "                    mode.\n",
      "                    \n",
      "    -j --java       Provides the full path to the java binary you want to use to\n",
      "                    launch fastqc. If not supplied then java is assumed to be in\n",
      "                    your path.\n",
      "                   \n",
      "    --noextract     Do not uncompress the output file after creating it.  You\n",
      "                    should set this option if you do not wish to uncompress\n",
      "                    the output when running in non-interactive mode.\n",
      "                    \n",
      "    --nogroup       Disable grouping of bases for reads >50bp. All reports will\n",
      "                    show data for every base in the read.  WARNING: Using this\n",
      "                    option will cause fastqc to crash and burn if you use it on\n",
      "                    really long reads, and your plots may end up a ridiculous size.\n",
      "                    You have been warned!\n",
      "                    \n",
      "    --min_length    Sets an artificial lower limit on the length of the sequence\n",
      "                    to be shown in the report.  As long as you set this to a value\n",
      "                    greater or equal to your longest read length then this will be\n",
      "                    the sequence length used to create your read groups.  This can\n",
      "                    be useful for making directly comaparable statistics from \n",
      "                    datasets with somewhat variable read lengths.\n",
      "                    \n",
      "    -f --format     Bypasses the normal sequence file format detection and\n",
      "                    forces the program to use the specified format.  Valid\n",
      "                    formats are bam,sam,bam_mapped,sam_mapped and fastq\n",
      "                    \n",
      "    -t --threads    Specifies the number of files which can be processed\n",
      "                    simultaneously.  Each thread will be allocated 250MB of\n",
      "                    memory so you shouldn't run more threads than your\n",
      "                    available memory will cope with, and not more than\n",
      "                    6 threads on a 32 bit machine\n",
      "                  \n",
      "    -c              Specifies a non-default file which contains the list of\n",
      "    --contaminants  contaminants to screen overrepresented sequences against.\n",
      "                    The file must contain sets of named contaminants in the\n",
      "                    form name[tab]sequence.  Lines prefixed with a hash will\n",
      "                    be ignored.\n",
      "\n",
      "    -a              Specifies a non-default file which contains the list of\n",
      "    --adapters      adapter sequences which will be explicity searched against\n",
      "                    the library. The file must contain sets of named adapters\n",
      "                    in the form name[tab]sequence.  Lines prefixed with a hash\n",
      "                    will be ignored.\n",
      "                    \n",
      "    -l              Specifies a non-default file which contains a set of criteria\n",
      "    --limits        which will be used to determine the warn/error limits for the\n",
      "                    various modules.  This file can also be used to selectively \n",
      "                    remove some modules from the output all together.  The format\n",
      "                    needs to mirror the default limits.txt file found in the\n",
      "                    Configuration folder.\n",
      "                    \n",
      "   -k --kmers       Specifies the length of Kmer to look for in the Kmer content\n",
      "                    module. Specified Kmer length must be between 2 and 10. Default\n",
      "                    length is 7 if not specified.\n",
      "                    \n",
      "   -q --quiet       Supress all progress messages on stdout and only report errors.\n",
      "   \n",
      "   -d --dir         Selects a directory to be used for temporary files written when\n",
      "                    generating report images. Defaults to system temp directory if\n",
      "                    not specified.\n",
      "                    \n",
      "BUGS\n",
      "\n",
      "    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk\n",
      "    or in www.bioinformatics.babraham.ac.uk/bugzilla/\n",
      "                   \n",
      "    \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "fastqc -help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "ad224f7e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/babin/bgi_test_exome'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4b53fd91",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 8.4G\r\n",
      "-rw-r--r-- 1 babin babin 4.2G Jun 29 13:25 \u001b[0m\u001b[01;31mraw_62_1.fq.gz\u001b[0m\r\n",
      "-rw-r--r-- 1 babin babin 4.2G Jun 29 13:42 \u001b[01;31mraw_62_2.fq.gz\u001b[0m\r\n"
     ]
    }
   ],
   "source": [
    "ls -lh ./1_input_data/"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "70478f18",
   "metadata": {},
   "source": [
    "### 1. Running FastQC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "63d8f856",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "fastqc \\\n",
    "./1_input_data/raw_62_1.fq.gz \\\n",
    "./1_input_data/raw_62_2.fq.gz \\\n",
    "-q \\\n",
    "-t 2 \\\n",
    "-o ./2_fastqc_out/\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "da0ce8bb",
   "metadata": {},
   "source": [
    "#### FastQC plots explanation:\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1f5775f1",
   "metadata": {},
   "source": [
    "- https://dnacore.missouri.edu/PDF/FastQC_Manual.pdf\n",
    "- https://rtsf.natsci.msu.edu/genomics/tech-notes/fastqc-tutorial-and-faq/"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3153f37f",
   "metadata": {},
   "source": [
    "### 2. Trimming reads\n",
    "- we have poort \"Per base sequence content\"\n",
    "- so we cut out 10 bases from the head\n",
    "\n",
    "#### trimmomatic manual\n",
    "- http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf\n",
    "- http://www.usadellab.org/cms/?page=trimmomatic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "291d001d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 32 ./1_input_data/raw_62_1.fq.gz ./1_input_data/raw_62_2.fq.gz ./3_trim_out/forw_paired_62_1.fq.gz ./3_trim_out/forw_unpaired_62_1.fq.gz ./3_trim_out/rev_paired_62_2.fq.gz ./3_trim_out/rev_unpaired_62_2.fq.gz HEADCROP:10 MINLEN:36\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 48115979 Both Surviving: 48115979 (100.00%) Forward Only Surviving: 0 (0.00%) Reverse Only Surviving: 0 (0.00%) Dropped: 0 (0.00%)\n",
      "TrimmomaticPE: Completed successfully\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "trimmomatic \\\n",
    "PE \\\n",
    "-threads 32 \\\n",
    "./1_input_data/raw_62_1.fq.gz \\\n",
    "./1_input_data/raw_62_2.fq.gz \\\n",
    "./3_trim_out/forw_paired_62_1.fq.gz \\\n",
    "./3_trim_out/forw_unpaired_62_1.fq.gz \\\n",
    "./3_trim_out/rev_paired_62_2.fq.gz \\\n",
    "./3_trim_out/rev_unpaired_62_2.fq.gz \\\n",
    "HEADCROP:10 \\\n",
    "MINLEN:36\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "56a09de2",
   "metadata": {},
   "source": [
    "### 3. mapping reads on reference by bwa"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6826ea42",
   "metadata": {},
   "source": [
    "#### 3.1 indexing reference sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4016caca",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bwa_index] Pack FASTA... 61.34 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=6469669378, availableWord=467229400\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 99999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 199999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 299999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 399999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 499999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 599999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 699999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 799999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 899999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 999999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 110 iterations done. 1099999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 120 iterations done. 1199999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 130 iterations done. 1299999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 140 iterations done. 1399999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 150 iterations done. 1499999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 160 iterations done. 1599999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 170 iterations done. 1699999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 180 iterations done. 1799999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 190 iterations done. 1899999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 200 iterations done. 1999999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 210 iterations done. 2099999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 220 iterations done. 2199999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 230 iterations done. 2299999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 240 iterations done. 2399999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 250 iterations done. 2499999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 260 iterations done. 2599999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 270 iterations done. 2699999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 280 iterations done. 2799999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 290 iterations done. 2899999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 300 iterations done. 2999999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 310 iterations done. 3099999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 320 iterations done. 3199999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 330 iterations done. 3299999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 340 iterations done. 3399999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 350 iterations done. 3499999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 360 iterations done. 3599999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 370 iterations done. 3699999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 380 iterations done. 3799999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 390 iterations done. 3899999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 400 iterations done. 3999999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 410 iterations done. 4099999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 420 iterations done. 4199999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 430 iterations done. 4299999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 440 iterations done. 4399999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 450 iterations done. 4499999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 460 iterations done. 4599999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 470 iterations done. 4699999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 480 iterations done. 4799999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 490 iterations done. 4899999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 500 iterations done. 4999999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 510 iterations done. 5099999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 520 iterations done. 5199999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 530 iterations done. 5299999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 540 iterations done. 5399999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 550 iterations done. 5499999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 560 iterations done. 5599999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 570 iterations done. 5699999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 580 iterations done. 5799995314 characters processed.\n",
      "[BWTIncConstructFromPacked] 590 iterations done. 5893725330 characters processed.\n",
      "[BWTIncConstructFromPacked] 600 iterations done. 5977028802 characters processed.\n",
      "[BWTIncConstructFromPacked] 610 iterations done. 6051065138 characters processed.\n",
      "[BWTIncConstructFromPacked] 620 iterations done. 6116864834 characters processed.\n",
      "[BWTIncConstructFromPacked] 630 iterations done. 6175343746 characters processed.\n",
      "[BWTIncConstructFromPacked] 640 iterations done. 6227315922 characters processed.\n",
      "[BWTIncConstructFromPacked] 650 iterations done. 6273504946 characters processed.\n",
      "[BWTIncConstructFromPacked] 660 iterations done. 6314553810 characters processed.\n",
      "[BWTIncConstructFromPacked] 670 iterations done. 6351034130 characters processed.\n",
      "[BWTIncConstructFromPacked] 680 iterations done. 6383453906 characters processed.\n",
      "[BWTIncConstructFromPacked] 690 iterations done. 6412264658 characters processed.\n",
      "[BWTIncConstructFromPacked] 700 iterations done. 6437867682 characters processed.\n",
      "[BWTIncConstructFromPacked] 710 iterations done. 6460619698 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 715 iterations.\n",
      "[bwa_index] 5295.15 seconds elapse.\n",
      "[bwa_index] Update BWT... 31.45 sec\n",
      "[bwa_index] Pack forward-only FASTA... 45.51 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 2105.74 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index 0_ref/GRCh37_latest_genomic.fna.gz\n",
      "[main] Real time: 7690.730 sec; CPU: 7539.189 sec\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "bwa index 0_ref/GRCh37_latest_genomic.fna.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b978430e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;31mforw_paired_62_1.fq.gz\u001b[0m    \u001b[01;31mrev_paired_62_2.fq.gz\u001b[0m\r\n",
      "\u001b[01;31mforw_unpaired_62_1.fq.gz\u001b[0m  \u001b[01;31mrev_unpaired_62_2.fq.gz\u001b[0m\r\n"
     ]
    }
   ],
   "source": [
    "ls 3_trim_out/"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5f6e1e35",
   "metadata": {},
   "source": [
    "#### 3.2 mapping reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c98884fc",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 674003, 26, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (231.94, 55.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (103, 206, 504)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1306)\n",
      "[M::mem_pestat] mean and std.dev: (267.09, 267.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1707)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 578.249 CPU sec, 35.902 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 673839, 15, 6)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (92, 239, 857)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2387)\n",
      "[M::mem_pestat] mean and std.dev: (301.50, 295.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3152)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.37, 55.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (137, 177, 962)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2612)\n",
      "[M::mem_pestat] mean and std.dev: (323.62, 376.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3437)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 573.430 CPU sec, 35.328 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 675423, 23, 6)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (172, 188, 270)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 466)\n",
      "[M::mem_pestat] mean and std.dev: (198.91, 73.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 564)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.31, 55.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (102, 173, 436)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1104)\n",
      "[M::mem_pestat] mean and std.dev: (244.45, 215.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1438)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 579.660 CPU sec, 35.741 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 671181, 9, 14)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.13, 55.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 136, 201)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 461)\n",
      "[M::mem_pestat] mean and std.dev: (122.33, 60.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 591)\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 593.291 CPU sec, 36.537 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 676541, 18, 9)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (194, 227, 268)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (46, 416)\n",
      "[M::mem_pestat] mean and std.dev: (232.75, 55.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 490)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 195, 514)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1310)\n",
      "[M::mem_pestat] mean and std.dev: (252.81, 233.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1708)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 575.790 CPU sec, 35.464 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 677055, 18, 13)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.45, 55.69)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (107, 150, 315)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 731)\n",
      "[M::mem_pestat] mean and std.dev: (175.87, 147.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 939)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (93, 170, 232)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 510)\n",
      "[M::mem_pestat] mean and std.dev: (153.92, 67.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 649)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 577.569 CPU sec, 35.591 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 675489, 11, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.30, 55.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 155, 193)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 343)\n",
      "[M::mem_pestat] mean and std.dev: (116.67, 69.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 418)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 588.329 CPU sec, 36.298 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 676556, 18, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 167, 183)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 317)\n",
      "[M::mem_pestat] mean and std.dev: (147.70, 39.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 384)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.38, 55.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (70, 300, 346)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 898)\n",
      "[M::mem_pestat] mean and std.dev: (217.38, 172.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1174)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 579.719 CPU sec, 35.674 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 677008, 12, 10)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (102, 169, 201)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 399)\n",
      "[M::mem_pestat] mean and std.dev: (152.67, 58.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 498)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.21, 55.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (90, 178, 263)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 609)\n",
      "[M::mem_pestat] mean and std.dev: (170.33, 92.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 782)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (181, 803, 980)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2578)\n",
      "[M::mem_pestat] mean and std.dev: (725.00, 705.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3546)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 581.110 CPU sec, 35.791 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 675409, 9, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (231.86, 55.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 607.017 CPU sec, 37.403 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 674310, 13, 12)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (125, 164, 3741)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10973)\n",
      "[M::mem_pestat] mean and std.dev: (1248.45, 1756.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 14589)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.19, 55.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (78, 155, 202)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 450)\n",
      "[M::mem_pestat] mean and std.dev: (135.54, 69.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 574)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (99, 138, 786)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2160)\n",
      "[M::mem_pestat] mean and std.dev: (350.18, 544.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2847)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 647.491 CPU sec, 39.845 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 674856, 12, 10)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.27, 55.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 127, 340)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 876)\n",
      "[M::mem_pestat] mean and std.dev: (150.36, 117.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1144)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 174, 233)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 467)\n",
      "[M::mem_pestat] mean and std.dev: (159.00, 61.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 584)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 601.117 CPU sec, 36.994 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 675400, 16, 9)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (119, 147, 166)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (25, 260)\n",
      "[M::mem_pestat] mean and std.dev: (133.22, 33.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 307)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.43, 55.76)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (88, 114, 180)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 364)\n",
      "[M::mem_pestat] mean and std.dev: (125.33, 77.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 456)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 609.331 CPU sec, 37.435 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 676119, 10, 13)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.00, 55.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (98, 769, 971)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2717)\n",
      "[M::mem_pestat] mean and std.dev: (398.25, 390.92)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3590)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (99, 158, 822)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2268)\n",
      "[M::mem_pestat] mean and std.dev: (319.54, 337.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2991)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 601.162 CPU sec, 37.016 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 676276, 15, 9)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.29, 55.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (147, 281, 806)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2124)\n",
      "[M::mem_pestat] mean and std.dev: (420.80, 355.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2783)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 587.232 CPU sec, 36.136 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 675937, 10, 10)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.26, 55.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (129, 157, 481)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1185)\n",
      "[M::mem_pestat] mean and std.dev: (350.00, 334.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1690)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (104, 166, 217)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 443)\n",
      "[M::mem_pestat] mean and std.dev: (146.78, 52.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 556)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 610.512 CPU sec, 37.444 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 676106, 11, 11)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.40, 55.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 168, 319)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 725)\n",
      "[M::mem_pestat] mean and std.dev: (165.50, 92.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 928)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 129, 199)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 361)\n",
      "[M::mem_pestat] mean and std.dev: (158.09, 64.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 442)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 603.012 CPU sec, 37.032 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 676245, 15, 10)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.40, 55.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (122, 213, 400)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 956)\n",
      "[M::mem_pestat] mean and std.dev: (202.15, 104.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1234)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (86, 108, 140)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 248)\n",
      "[M::mem_pestat] mean and std.dev: (123.70, 53.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 336)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 614.483 CPU sec, 37.754 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 674133, 15, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.34, 55.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (114, 156, 315)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 717)\n",
      "[M::mem_pestat] mean and std.dev: (142.92, 83.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 918)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 615.657 CPU sec, 37.863 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 674189, 20, 9)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.02, 55.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 147, 204)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 390)\n",
      "[M::mem_pestat] mean and std.dev: (146.28, 77.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 483)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 620.384 CPU sec, 38.103 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 676047, 16, 8)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (97, 126, 257)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 577)\n",
      "[M::mem_pestat] mean and std.dev: (143.78, 68.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 737)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.14, 55.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (124, 182, 938)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2566)\n",
      "[M::mem_pestat] mean and std.dev: (356.07, 404.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3380)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 598.715 CPU sec, 36.736 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 676196, 17, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (194, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (48, 413)\n",
      "[M::mem_pestat] mean and std.dev: (232.33, 55.42)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 486)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (104, 189, 1853)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5351)\n",
      "[M::mem_pestat] mean and std.dev: (454.00, 715.81)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7100)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 589.944 CPU sec, 36.218 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 676653, 10, 7)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (92, 170, 300)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 716)\n",
      "[M::mem_pestat] mean and std.dev: (170.75, 78.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 924)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.10, 55.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (174, 213, 320)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 612)\n",
      "[M::mem_pestat] mean and std.dev: (206.78, 89.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 758)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 620.368 CPU sec, 38.151 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 674785, 14, 10)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.35, 55.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (47, 117, 167)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 407)\n",
      "[M::mem_pestat] mean and std.dev: (102.23, 60.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 527)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (113, 130, 186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 332)\n",
      "[M::mem_pestat] mean and std.dev: (142.11, 60.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 405)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 588.563 CPU sec, 36.143 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 672380, 12, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.24, 55.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (90, 171, 324)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 792)\n",
      "[M::mem_pestat] mean and std.dev: (148.40, 117.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1026)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 587.718 CPU sec, 36.157 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 676213, 13, 13)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (107, 139, 4114)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 12128)\n",
      "[M::mem_pestat] mean and std.dev: (1339.70, 1858.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 16135)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.32, 55.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 148, 400)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1058)\n",
      "[M::mem_pestat] mean and std.dev: (157.82, 133.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1387)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (124, 132, 256)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 520)\n",
      "[M::mem_pestat] mean and std.dev: (149.91, 58.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 652)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 584.242 CPU sec, 35.942 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (14, 675405, 15, 6)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (105, 157, 234)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 492)\n",
      "[M::mem_pestat] mean and std.dev: (150.75, 56.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 621)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.32, 55.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (124, 162, 551)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1405)\n",
      "[M::mem_pestat] mean and std.dev: (308.50, 308.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1832)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 589.214 CPU sec, 36.193 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 675861, 13, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.33, 55.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (82, 168, 432)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1132)\n",
      "[M::mem_pestat] mean and std.dev: (241.92, 277.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1482)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 589.852 CPU sec, 36.387 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 672624, 18, 12)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.17, 55.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (155, 350, 1001)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2693)\n",
      "[M::mem_pestat] mean and std.dev: (424.20, 338.26)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3539)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (104, 158, 227)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 473)\n",
      "[M::mem_pestat] mean and std.dev: (163.55, 82.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 596)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 598.338 CPU sec, 36.825 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 674620, 13, 11)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 266)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (47, 412)\n",
      "[M::mem_pestat] mean and std.dev: (231.71, 55.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 485)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (136, 227, 396)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 916)\n",
      "[M::mem_pestat] mean and std.dev: (200.18, 105.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1176)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (76, 112, 192)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 424)\n",
      "[M::mem_pestat] mean and std.dev: (98.78, 46.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 540)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 604.485 CPU sec, 37.142 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (14, 668038, 14, 7)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (82, 137, 175)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 361)\n",
      "[M::mem_pestat] mean and std.dev: (127.85, 56.43)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 454)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.03, 55.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (108, 176, 302)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 690)\n",
      "[M::mem_pestat] mean and std.dev: (146.55, 89.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 884)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 599.551 CPU sec, 36.854 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 676706, 20, 10)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (126, 172, 326)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 726)\n",
      "[M::mem_pestat] mean and std.dev: (172.88, 72.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 926)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.01, 55.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (137, 278, 355)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 791)\n",
      "[M::mem_pestat] mean and std.dev: (192.69, 110.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1009)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (97, 132, 193)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 385)\n",
      "[M::mem_pestat] mean and std.dev: (124.56, 56.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 481)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 596.353 CPU sec, 36.665 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 676598, 15, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (106, 148, 211)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 421)\n",
      "[M::mem_pestat] mean and std.dev: (135.89, 58.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 526)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.01, 55.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (112, 184, 387)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 937)\n",
      "[M::mem_pestat] mean and std.dev: (196.69, 138.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1212)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 599.004 CPU sec, 36.787 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 676426, 17, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (231.93, 55.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (100, 157, 948)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2644)\n",
      "[M::mem_pestat] mean and std.dev: (339.07, 415.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3492)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 581.119 CPU sec, 35.718 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 674639, 16, 14)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (231.90, 55.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (78, 127, 244)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 576)\n",
      "[M::mem_pestat] mean and std.dev: (125.36, 94.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 742)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (150, 230, 412)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 936)\n",
      "[M::mem_pestat] mean and std.dev: (242.33, 152.59)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1198)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 586.115 CPU sec, 36.012 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 675185, 19, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.04, 55.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (109, 191, 1550)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4432)\n",
      "[M::mem_pestat] mean and std.dev: (792.78, 1080.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5873)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 591.969 CPU sec, 36.431 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 677291, 19, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.23, 55.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (101, 177, 291)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 671)\n",
      "[M::mem_pestat] mean and std.dev: (158.94, 96.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 861)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 585.714 CPU sec, 36.013 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 676597, 25, 8)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (126, 157, 202)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 354)\n",
      "[M::mem_pestat] mean and std.dev: (145.82, 45.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 430)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.05, 55.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (117, 156, 243)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 495)\n",
      "[M::mem_pestat] mean and std.dev: (187.40, 99.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 621)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 593.107 CPU sec, 36.564 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 673773, 20, 12)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (231.94, 55.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (185, 315, 770)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1940)\n",
      "[M::mem_pestat] mean and std.dev: (398.89, 288.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2525)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (143, 175, 230)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 404)\n",
      "[M::mem_pestat] mean and std.dev: (157.82, 63.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 491)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 608.251 CPU sec, 37.485 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 672262, 14, 3)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.05, 55.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (123, 327, 922)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2520)\n",
      "[M::mem_pestat] mean and std.dev: (528.14, 645.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3319)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 594.312 CPU sec, 36.578 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 675248, 12, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.17, 55.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (75, 206, 430)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1140)\n",
      "[M::mem_pestat] mean and std.dev: (276.33, 267.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1495)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 574.243 CPU sec, 35.476 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 675748, 13, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.27, 55.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (95, 142, 232)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 506)\n",
      "[M::mem_pestat] mean and std.dev: (154.23, 78.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 643)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 572.915 CPU sec, 35.246 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 676183, 8, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (231.92, 55.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 576.556 CPU sec, 35.461 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (5, 675641, 23, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (231.98, 55.59)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (41, 89, 226)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 596)\n",
      "[M::mem_pestat] mean and std.dev: (119.82, 95.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 781)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 582.849 CPU sec, 35.888 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 675314, 10, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.18, 55.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (7, 120, 353)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1045)\n",
      "[M::mem_pestat] mean and std.dev: (178.90, 168.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1391)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 578.184 CPU sec, 35.578 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 674491, 11, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.34, 55.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (172, 313, 597)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1447)\n",
      "[M::mem_pestat] mean and std.dev: (402.45, 393.69)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1977)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 667.742 CPU sec, 41.249 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 675490, 16, 4)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (231.82, 55.54)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (168, 322, 1380)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3804)\n",
      "[M::mem_pestat] mean and std.dev: (603.20, 860.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5016)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 577.484 CPU sec, 35.615 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 675461, 19, 12)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.06, 55.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (80, 161, 954)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2702)\n",
      "[M::mem_pestat] mean and std.dev: (379.24, 495.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3576)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (133, 210, 836)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2242)\n",
      "[M::mem_pestat] mean and std.dev: (337.18, 300.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2945)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 586.214 CPU sec, 36.048 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 670972, 16, 9)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (231.92, 55.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (146, 195, 928)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2492)\n",
      "[M::mem_pestat] mean and std.dev: (341.43, 410.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3274)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 606.600 CPU sec, 37.444 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 671896, 14, 10)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (231.96, 55.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (33, 129, 186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 492)\n",
      "[M::mem_pestat] mean and std.dev: (118.07, 94.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 645)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (151, 163, 517)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1249)\n",
      "[M::mem_pestat] mean and std.dev: (388.40, 334.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1725)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 596.827 CPU sec, 36.831 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 674289, 17, 7)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.03, 55.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (71, 144, 319)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 815)\n",
      "[M::mem_pestat] mean and std.dev: (173.94, 132.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1063)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 598.115 CPU sec, 36.797 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 668968, 24, 11)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 266)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (47, 412)\n",
      "[M::mem_pestat] mean and std.dev: (231.44, 55.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 485)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (134, 179, 309)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 659)\n",
      "[M::mem_pestat] mean and std.dev: (166.19, 94.92)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 834)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 138, 178)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 302)\n",
      "[M::mem_pestat] mean and std.dev: (137.50, 66.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 403)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 615.275 CPU sec, 37.920 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 675327, 15, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (231.85, 55.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (142, 256, 932)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2512)\n",
      "[M::mem_pestat] mean and std.dev: (492.43, 618.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3302)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 601.872 CPU sec, 37.050 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (7, 675506, 20, 13)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (231.92, 55.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (58, 172, 241)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 607)\n",
      "[M::mem_pestat] mean and std.dev: (150.84, 107.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 790)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (96, 130, 240)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 528)\n",
      "[M::mem_pestat] mean and std.dev: (131.45, 49.42)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 672)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 593.453 CPU sec, 36.532 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 674825, 19, 10)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 227, 267)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (45, 415)\n",
      "[M::mem_pestat] mean and std.dev: (232.31, 55.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 489)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (100, 175, 797)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2191)\n",
      "[M::mem_pestat] mean and std.dev: (389.42, 395.85)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2888)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (179, 239, 378)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 776)\n",
      "[M::mem_pestat] mean and std.dev: (195.62, 90.59)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 975)\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 593.273 CPU sec, 36.560 real sec\n",
      "[M::process] read 1684212 sequences (160000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (16, 674399, 12, 6)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (92, 166, 211)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 449)\n",
      "[M::mem_pestat] mean and std.dev: (136.71, 57.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 568)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 266)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (47, 412)\n",
      "[M::mem_pestat] mean and std.dev: (231.40, 55.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 485)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (120, 315, 329)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 747)\n",
      "[M::mem_pestat] mean and std.dev: (231.17, 136.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 956)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 618.943 CPU sec, 38.206 real sec\n",
      "[M::process] read 231874 sequences (22028030 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 671537, 8, 11)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (193, 226, 266)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (47, 412)\n",
      "[M::mem_pestat] mean and std.dev: (231.29, 55.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 485)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (134, 172, 916)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2480)\n",
      "[M::mem_pestat] mean and std.dev: (372.45, 368.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3262)\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1684212 reads in 621.980 CPU sec, 38.632 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 92095, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (192, 225, 265)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (46, 411)\n",
      "[M::mem_pestat] mean and std.dev: (230.68, 55.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 484)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 231874 reads in 98.017 CPU sec, 6.045 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 16 0_ref/GRCh37_latest_genomic.fna.gz 3_trim_out/forw_paired_62_1.fq.gz 3_trim_out/rev_paired_62_2.fq.gz\n",
      "[main] Real time: 2215.663 sec; CPU: 34109.326 sec\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "bwa mem \\\n",
    "-t 16 \\\n",
    "0_ref/GRCh37_latest_genomic.fna.gz \\\n",
    "3_trim_out/forw_paired_62_1.fq.gz \\\n",
    "3_trim_out/rev_paired_62_2.fq.gz > \\\n",
    "4_bwa_out/bgi_exome.sam "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "135ef653",
   "metadata": {},
   "source": [
    "### 4. dealing with bam file\n",
    "\n",
    "- manual : http://www.htslib.org/workflow/ (see \"WGS/WES Mapping to Variant Calls - Version 1.0\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ead6af68",
   "metadata": {},
   "source": [
    "#### 4.1 fixmate : clean up read pairing information and flags after bwa mapping\n",
    "- `-O` output format\n",
    "- `--threads` number of threads to use\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "93d0d4db",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "samtools fixmate -O bam \\\n",
    "--threads 16 \\\n",
    "4_bwa_out/bgi_exome.sam \\\n",
    "4_bwa_out/bgi_exome_fixmate_62.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ed046aa3",
   "metadata": {},
   "source": [
    "#### 4.2 sort  reads by genome position\n",
    "- `-O` output format\n",
    "- `--threads` number of threads to use\n",
    "- `-m` memory for each thread in Gb\n",
    "- `-o` output file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "92745666",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bam_sort_core] merging from 24 files and 8 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "samtools sort \\\n",
    "-O bam \\\n",
    "--threads 8 \\\n",
    "-m 1G \\\n",
    "-o ./4_bwa_out/bgi_exome_fixmate_sorted_62.bam \\\n",
    "./4_bwa_out/bgi_exome_fixmate_62.bam \n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3adc61f1",
   "metadata": {},
   "source": [
    "#### 4.3 indexing sorted bam\n",
    "- `-@` number of threads to use"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "fa1dd6e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "samtools index \\\n",
    "./4_bwa_out/bgi_exome_fixmate_sorted_62.bam \\\n",
    "-@ 16"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "14752c5f",
   "metadata": {},
   "source": [
    "### 5. bcftools variant calling\n",
    "\n",
    "- `mpileup` generates genotype likelihoods\n",
    "- `call`  makes actuall calling\n",
    "- `-f`  reference file\n",
    "- `-m`  default calling method\n",
    "- `-v`  output only variant sites\n",
    "- `-O`  selecting output format 'b' compressed BCF; 'u' uncompressed BCF; 'z' compressed VCF; 'v' uncompressed VCF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "aac1dc8c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "bcftools mpileup \\\n",
    "-f 0_ref/GRCh37_latest_genomic.fna \\\n",
    "./4_bwa_out/bgi_exome_fixmate_sorted_62.bam \\\n",
    "--threads 16 \\\n",
    "| \\\n",
    "bcftools call \\\n",
    "-mv -Ov \\\n",
    "-o ./5_variants/bgi_test.vcf \\\n",
    "--threads 16"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2df60e8d",
   "metadata": {},
   "source": [
    "### 7. annotating our vcf with rs IDs from dbsnp vcf "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "71b88056",
   "metadata": {},
   "source": [
    "#### 7.1 compressing vcf file prior to annotation "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "4edf6658",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "bgzip -c \\\n",
    "./6_variants_annot_with_dbsnp/bgi_test.vcf > \\\n",
    "./6_variants_annot_with_dbsnp/bgi_test.vcf.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8ecb972f",
   "metadata": {},
   "source": [
    "#### 7.2 indexing compressed vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "cfd0b650",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "tabix -p \\\n",
    "vcf \\\n",
    "./6_variants_annot_with_dbsnp/bgi_test.vcf.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "75ecec8f",
   "metadata": {},
   "source": [
    "#### 7.2 indexing dbsnp vcf "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "e45376e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "tabix -p \\\n",
    "vcf \\\n",
    "./0_annot_dbsnp_37/GRCh37_latest_dbSNP_all.vcf.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "db18cfd3",
   "metadata": {},
   "source": [
    "#### 7.3 annotating our vcf with rs IDs from dbsnp\n",
    "- `-a` file with annotations\n",
    "- `-c` columns to take from file with annotations\n",
    "- `-o` output file with annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "313d6e4f",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "bcftools annotate \\\n",
    "-a ./0_annot_dbsnp_37/GRCh37_latest_dbSNP_all.vcf.gz \\\n",
    "-c ID \\\n",
    "-o ./6_variants_annot_with_dbsnp/bgi_test_annotated.vcf.gz \\\n",
    "./6_variants_annot_with_dbsnp/bgi_test.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3a451b73",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "860f4a54",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ae43ed67",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8f232f22",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b7fd7a0b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39731aee",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6ce536fe",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3011affc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fec307b5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8328b15d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
